transport/transporters/kafka/batch/batch.go

```diff
@@ -49,10 +62,17 @@ func (b *KafkaBatch) Add(msg *marshaller.MarshalledMessage) (bool, error) {
 		return false, errors.New("batch is full")
 	}
 
+	bytes := []byte(msg.PartitionKey)
+	hasher := fnv.New32a()
+	_, err := hasher.Write(bytes)
+	if err != nil {
+		return false, err
+	}
+	partition := int32(hasher.Sum32()&0x7fffffff) % b.kafkaPartitionCount
 	kafkaMsg := &sarama.ProducerMessage{
-		Topic: b.topic,
-		Value: sarama.ByteEncoder(msg.Json),
-		Key:   sarama.StringEncoder(msg.PartitionKey),
+		Topic:     b.topic,
+		Value:     sarama.ByteEncoder(msg.Json),
+		Partition: partition,
```

Comment:
hashing the partition key and modding the value by the # of available partition. A transaction is written to the same kafka partition as a batch.